core/camel-api/src/main/java/org/apache/camel/Exchange.java

```diff
@@ -117,6 +117,7 @@ public interface Exchange extends VariableAware {
     String CONTENT_TYPE = "Content-Type";
     String COOKIE_HANDLER = "CamelCookieHandler";
     String CORRELATION_ID = "CamelCorrelationId";
+    String ATTACHMENTS_SIZE = "CamelAttachmentsSize";
```

Comment:
I think there is somewhere you need to document this header, try look for some of the other headers.

Reply:
Also put it up to A headers so its sorted

Reply:
Ah yeah add it to `ExchangePropertyKey` also and doc via `@Metadata`

Reply:
mmm, I cannot find much documentation for other headers, should I annotate it with Metadata and add a description?

Reply:
Yes

Reply:
https://github.com/apache/camel/pull/17070/files#diff-c4832886119fc6dd6a4a54e68c5e2740ea9ae86b8699a469648284189d51e839 like this? I do not fully understand how `ExchangePropertyKey` works, for example, FILE* related headers are not there.

Reply:
Its the most commonly used headers - and since HTTP with attachments is fairly common then IMHO it would be okay to add here